src/Core/Domain/Product/Command/AddCategoryToProductCommand.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+
+/**
+ * Class AddCategoryToProductCommand adds a product to a category.
+ */
+class AddCategoryToProductCommand
+{
+    /**
+     * @var int
+     */
+    private $categoryId;
+
+    /**
+     * @var int
+     */
+    private $productId;
+
+    /**
+     * @param int $categoryId
+     * @param int $productId
+     *
+     * @throws CategoryConstraintException
+     * @throws ProductConstraintException
+     */
+    public function __construct($categoryId, $productId)
+    {
+        $this->setCategoryId($categoryId);
+        $this->setProductId($productId);
```

Comment:
```suggestion
        $this->productId = new ProductId($productId);
```

You need here to create the ValueObject ProductId (see VO CategoryId as an example) and use it 😉 
and you can move your validation logic into the ValueObject to make it reusable

Reply:
done !